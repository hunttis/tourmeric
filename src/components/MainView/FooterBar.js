import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import _ from 'lodash';
import PropTypes from 'prop-types';

import PrivacyPolicy from './PrivacyPolicy/PrivacyPolicy-container';

export default class FooterBar extends Component {

  foo() {}

  render() {

    const { settings } = this.props;

    const showingSponsors = _.get(settings, 'showSponsors', false);
    const hasAtLeastOneSponsor = showingSponsors && !_.isEmpty(_.get(settings, 'footer', {}));

    return (
      <footer className="footer less-bottompadding">
        <div className="content">
          <div className="columns">

            {hasAtLeastOneSponsor &&
            <Fragment>
              <div className="column has-text-centered is-hidden-desktop">
                <Translate id="sponsoredby" />:
              </div>
              <div className="column has-text-left is-hidden-mobile">
                <Translate id="sponsoredby" />:
              </div>
            </Fragment>
            }
            {(showingSponsors && _.get(settings, 'footer.first.image')) &&
            <div className="column is-vcentered">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.first.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.first.image} alt="" />
                </a>
              </figure>
            </div>
            }
            {(showingSponsors && _.get(settings, 'footer.second.image')) &&
            <div className="column">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.second.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.second.image} alt="" />
                </a>
              </figure>
            </div>
            }
            {(showingSponsors && _.get(settings, 'footer.third.image')) &&
            <div className="column">
              <figure className="image is-paddingless is-marginless">
                <a href={settings.footer.third.link} target="_blank" rel="noopener noreferrer">
                  <img className="footerImage" src={settings.footer.third.image} alt="" />
                </a>
              </figure>
            </div>
            }
            <div className="column has-text-centered is-hidden-desktop">
              <PrivacyPolicy showAcceptance={false} />
            </div>
            <div className="column has-text-right is-hidden-mobile">
              <PrivacyPolicy showAcceptance={false} />
            </div>
          </div>
        </div>
      </footer>
    );
  }
}

FooterBar.propTypes = {
  settings: PropTypes.object,
};
