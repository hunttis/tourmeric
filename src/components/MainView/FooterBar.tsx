import React from 'react';
import { FormattedMessage, IntlShape } from "react-intl";
import _ from 'lodash';
import { isLoaded } from 'react-redux-firebase';
import PrivacyPolicy from './PrivacyPolicy/PrivacyPolicy-container';
import { Settings } from '../../models/Settings';

interface Props {
  settings: Settings;
}

const FooterBar = ({ settings }: Props) => {

  const showingSponsors = _.get(settings, 'showSponsors', false);
  const hasAtLeastOneSponsor = showingSponsors && !_.isEmpty(_.get(settings, 'footer', {}));

  if (!isLoaded(settings)) {
    return <div />;
  }

  return (
    <footer className="footer less-bottompadding">
      <div className="content">
        <div className="columns is-multiline">

          {hasAtLeastOneSponsor &&
            <>
              <div className="column is-12 has-text-centered is-hidden-tablet">
                <FormattedMessage id="sponsoredby" />:
              </div>
              <div className="column is-6 has-text-left is-hidden-mobile">
                <FormattedMessage id="sponsoredby" />:
              </div>
              <div className="column is-6 has-text-right is-hidden-mobile">
                <PrivacyPolicy showAcceptance={false} />
              </div>
            </>
          }

          {showingSponsors &&
            <>
              {settings.footer && Object.entries(settings.footer).map((footerEntry) => {

                const key = footerEntry[0];
                const value = footerEntry[1];

                return (
                  <div key={`Footer-${key}`} className="column sponsor">
                    <figure className="image is-paddingless is-marginless">
                      <a href={value.link} target="_blank" rel="noopener noreferrer" title={value.text}>
                        <img className="footerImage" src={value.image} alt="" />
                      </a>
                    </figure>
                  </div>
                );
              })}
            </>
          }

          <div className="column is-12 has-text-centered is-hidden-tablet">
            <PrivacyPolicy showAcceptance={false} />
          </div>

        </div>
      </div>
    </footer>
  );
};

export default FooterBar;
