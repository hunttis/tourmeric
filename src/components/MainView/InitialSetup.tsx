import React, { Component } from 'react';
import { Translate } from 'react-localize-redux';
import { isLoaded, isEmpty } from 'react-redux-firebase';

import { registerEmail } from '../../api/loginApi';
import { FirebaseProfile } from '~/models/ReduxState';

interface Props {
  profile: FirebaseProfile;
}

interface State {
  loginEmail: string;
  loginPass: string;
  errorState: boolean;
}

export default class InitialSetup extends Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.state = { loginEmail: '', loginPass: '', errorState: false };
  }

  onChangeEmail = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ loginEmail: event.target.value });
  }

  onChangePass = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({ loginPass: event.target.value });
  }

  onRegisterSubmit = async () => {
    try {
      await registerEmail(this.state.loginEmail, this.state.loginPass);
      this.setState({ errorState: false });
    } catch (err) {
      this.setState({ errorState: err.code });
    }
  }

  render() {
    const { profile } = this.props;
    const showSetupEnd = isLoaded(profile) && !isEmpty(profile);

    return (
      <div className="section">
        <div className="columns is-multiline">
          {!showSetupEnd &&
            <>
              <div className="column is-12">
                <h1 className="title">
                  <Translate id="setupnotdoneyet" />
                </h1>
              </div>
              <div className="column is-12">
                <p>
                  <Translate id="youneedtocreateanadminaccount" />
                </p>
              </div>
              <div className="column is-12">
                <h2 className="subtitle"><Translate id="fillinyouremailaddressandinputpassword" /></h2>
              </div>

              <div className="column is-12">
                <div className="field is-grouped">
                  <p className="control is-expanded has-icons-left">
                    <span className="icon is-small is-left"><i className="fas fa-envelope" /></span>
                    <Translate>
                      {({ translate }) => (
                        <input placeholder={`${translate('emailplaceholder')}`} className="input email" type="email" onChange={(event) => this.onChangeEmail(event)} />
                      )}
                    </Translate>
                  </p>
                  <p className="control is-expanded has-icons-left">
                    <span className="icon is-small is-left"><i className="fas fa-lock" /></span>
                    <Translate>
                      {({ translate }) => (
                        <input placeholder={`${translate('passwordplaceholder')}`} className="input password" type="password" onChange={(event) => this.onChangePass(event)} />
                      )}
                    </Translate>
                  </p>
                  <button className="button" onClick={() => this.onRegisterSubmit()}><Translate id="register" /></button>
                </div>
              </div>
            </>
          }
          {this.state.errorState &&
            <>
              <div className="column is-12">
                <h1 className="title">
                  <Translate id="almostthere" />
                </h1>
              </div>
              <div className="column is-12">
                <div className="box">
                  <Translate id={String(this.state.errorState)} />
                </div>
              </div>
            </>
          }
          {showSetupEnd &&
            <div className="column is-12">
              <h2 className="subtitle">
                Alright, time to finish the setup. Next part is a bit tricky.
              </h2>
              <ol className="list">
                <li>You need to go into the <a href="https://console.firebase.google.com">Firebase console (click here to be taken there)</a></li>
                <li>
                  Select the <span className="has-text-success">app you created</span> there. Then click <span className="has-text-success">database</span>.
                </li>
                <li>
                  Under the users folder, you should see one user. It&apos;s the one you just registered. Click the ID of that user (it&apos;s a garbled mess of characters and numbers)
                </li>
                <li>
                  Once you see the variables for that user, you should see an email address and a username for it. Point at the id and press the <span className="has-text-danger">+</span>
                </li>
                <li>
                  For the <span className="has-text-success">name</span> write <span className="has-text-danger">role</span> and for the <span className="has-text-success">value</span> write <span className="has-text-danger">admin</span>. This makes your user account an admin account.
                </li>
                <li>
                  Go back. You can press the back button on the browser or click the name of your application just above the user ID.
                </li>
                <li>
                  Point at the <span className="has-text-success">database root</span> (it&apos;s probably named what your application is named) with your mouse cursor and click the <span className="has-text-danger">+</span> to create a new folder.
                </li>
                <li>
                  For the <span className="has-text-success">name</span> write <span className="has-text-danger">settings</span> and click the <span className="has-text-danger">+</span> next to the value. It creates a variable under the settings folder.
                </li>
                <li>
                  For the new variable: <span className="has-text-success">name</span> should be <span className="has-text-danger">initialSetupDone</span> and the <span className="has-text-success">value</span> for it should be <span className="has-text-danger">true</span>.
                </li>
                <li>
                  Once you refresh the page, you should see the user info page. Input your info there and start using tourmeric!
                </li>
              </ol>

            </div>
          }
        </div>
      </div>
    );
  }
}
