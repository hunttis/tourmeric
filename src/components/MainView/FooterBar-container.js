import { connect } from 'react-redux';
import { compose } from 'redux';

import FooterBar from './FooterBar';

export default compose(
  connect(state => ({
    settings: state.firebase.data.settings,
  })),
  connect(({ firebase: { auth, profile } }) => ({ auth, profile })),
)(FooterBar);
