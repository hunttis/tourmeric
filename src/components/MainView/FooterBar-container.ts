import { connect } from 'react-redux';
import { compose } from 'redux';

import FooterBar from './FooterBar';
import { ReduxState } from '~/models/ReduxState';

export default compose(
  connect((state: ReduxState) => ({
    settings: state.firebase.data.settings,
  })),
  connect(({ firebase: { auth, profile } }: ReduxState) => ({ auth, profile })),
)(FooterBar) as React.ComponentType<any>;
