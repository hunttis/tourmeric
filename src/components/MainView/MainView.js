import React, { Component } from 'react';
import { isLoaded, isEmpty } from 'react-redux-firebase';
import _ from 'lodash';
import { Translate, setActiveLanguage } from 'react-localize-redux';
import Moment from 'react-moment';
import PropTypes from 'prop-types';
import AdminTools from '../AdminTools/AdminTools-container';
import AdminToolsEvents from '../AdminTools/AdminToolsEvents-container';
import AdminSiteSettings from '../AdminTools/SiteSettings/AdminSiteSettings-container';
import UserInfo from '../UserInfo/UserInfo-container';
import EventList from '../EventList/EventList-container';
import TitleBar from './TitleBar-container';
import Login from './Account/Login-container';
import Register from './Account/Register-container';
import ThemeHandler from './ThemeHandler-container';
import StoreInfo from '../StoreInfo/StoreInfo-container';
import Navbar from './Navbar/Navbar-container';
import Today from './Today/Today-container';
import CompanyInfo from './CompanyInfo/CompanyInfo-container';
import InitialSetup from './InitialSetup';
import FooterBar from './FooterBar';

// ******************
// Set the theme here
// ******************
import 'bulmaswatch/darkly/bulmaswatch.min.css';

export default class MainView extends Component {
  constructor(props) {
    super(props);
    this.changeLanguage = this.changeLanguage.bind(this);
    this.switchActiveTab = this.switchActiveTab.bind(this);
    this.state = { forceUserInfo: false };
  }

  componentWillReceiveProps(nextProps) {
    const isProfileLoaded = isLoaded(nextProps.profile);
    const isLoggedIn = isProfileLoaded && !isEmpty(nextProps.profile);
    const acceptedPrivacyPolicy = _.get(nextProps.profile, 'acceptedPrivacyPolicy', false);

    if (isLoggedIn && (!nextProps.profile.firstName || !nextProps.profile.lastName || !nextProps.profile.email || !acceptedPrivacyPolicy)) {
      this.setState({ forceUserInfo: true, activeItem: 'userinfo' });
    } else if (isLoaded(nextProps.profile)) {
      this.setState({ forceUserInfo: false });
    }
  }

  switchActiveTab(type) {
    this.setState({ activeItem: type });
  }

  changeLanguage(newLanguage) {
    const { dispatch } = this.props;
    dispatch(setActiveLanguage(newLanguage));
    Moment.globalLocale = newLanguage;
  }

  render() {
    const { activeItem, forceUserInfo } = this.state;
    const { profile, settings } = this.props;

    if (isLoaded(settings) && isEmpty(settings)) {
      return <InitialSetup profile={profile} />;
    }

    const isProfileLoaded = isLoaded(profile);

    if (isProfileLoaded) {
      const isLoggedIn = isProfileLoaded && !isEmpty(profile);
      const isAdmin = isLoggedIn && _.get(profile, 'role', 'user') === 'admin';
      const hasProfileData = isProfileLoaded && profile.firstName && profile.lastName && profile.email;

      const eventsActive = _.get(settings, 'features.events.active', false);
      const storeInfoActive = _.get(settings, 'features.storeinfo.active', false);

      const activePage = activeItem || _.get(profile, 'landingPage', 'today');

      const todayVisible = Boolean(!forceUserInfo && (!isLoggedIn || hasProfileData) && activePage === 'today');
      const eventContentVisible = Boolean(!forceUserInfo && (!isLoggedIn || hasProfileData) && activePage === 'events');
      const userInfoVisible = Boolean(isLoggedIn && (forceUserInfo || activePage === 'userinfo' || !hasProfileData));
      const adminToolsVisible = Boolean(isAdmin && !forceUserInfo && activePage === 'admintools');
      const adminToolsEventsVisible = Boolean(isAdmin && !forceUserInfo && activePage === 'admintoolsevents');
      const adminSiteSettingsVisible = Boolean(isAdmin && !forceUserInfo && activePage === 'adminsitesettings');
      const loginVisible = Boolean(!isLoggedIn && activePage === 'login');
      const registerVisible = Boolean(!isLoggedIn && activePage === 'register');
      const storeInfoVisible = Boolean(activePage === 'storeinfo');
      const companyInfoVisible = Boolean(activePage === 'companyinfo');

      return (
        <div>
          <ThemeHandler />
          <TitleBar returnToFrontpage={() => this.switchActiveTab('today')} />
          <Navbar switchActiveTab={this.switchActiveTab} activeItem={activePage} changeLanguage={this.changeLanguage} />
          {todayVisible && <Today />}
          {eventsActive && eventContentVisible && <EventList />}
          {storeInfoActive && storeInfoVisible && <StoreInfo />}
          {userInfoVisible && <UserInfo />}
          {adminToolsVisible && <AdminTools />}
          {adminToolsEventsVisible && <AdminToolsEvents />}
          {adminSiteSettingsVisible && <AdminSiteSettings />}
          {loginVisible && <Login />}
          {registerVisible && <Register />}
          {companyInfoVisible && <CompanyInfo />}
          {isLoaded(settings) &&
            <FooterBar />
          }
        </div>

      );
    }
    return (
      <div className="title has-text-centered">
        <div className="level" />
        <button disabled className="button is-loading is-black">Loading..</button>
      </div>
    );
  }
}

const MainViewTab = ({ isActive, switchAction, icon, translationKey, notification, isDisabled }) => (
  <li className={`has-icon ${isActive && 'is-active'} ${isDisabled && 'has-text-gray'}`}>
    <a onClick={switchAction}>
      <span className="icon is-small"><i className={`fas ${icon}`} aria-hidden="true" /></span>
      <span><Translate id={translationKey} /></span>
      {notification &&
        <span className="icon is-small has-text-danger"><i className={`fas ${notification}`} aria-hidden="true" /></span>
      }
    </a>
  </li>
);

MainViewTab.propTypes = {
  isActive: PropTypes.bool,
  switchAction: PropTypes.func,
  icon: PropTypes.string,
  translationKey: PropTypes.string,
  notification: PropTypes.string,
  isDisabled: PropTypes.bool,
};

MainView.propTypes = {
  dispatch: PropTypes.func,
  profile: PropTypes.object,
  settings: PropTypes.object,
};
