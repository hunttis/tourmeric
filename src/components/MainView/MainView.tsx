import React, { Component } from 'react';
import { Dispatch } from 'redux';
import { isLoaded, isEmpty } from 'react-redux-firebase';
import _ from 'lodash';
import { setActiveLanguage } from 'react-localize-redux';
import Moment from 'react-moment';
import { Route, Switch } from 'react-router-dom';
import { Location, History } from 'history';

import AdminTools from '../AdminTools/AdminTools-container';
import AdminToolsEvents from '../AdminTools/AdminToolsEvents-container';
import AdminSiteSettings from '../AdminTools/SiteSettings/AdminSiteSettings-container';
import UserInfo from '../UserInfo/UserInfo-container';
import EventCalendar from '../EventList/EventCalendar/EventCalendar-container';
import TitleBar from './TitleBar-container';
import Login from './Account/Login-container';
import Register from './Account/Register-container';
import ThemeHandler from './ThemeHandler-container';
import StoreInfo from '../StoreInfo/StoreInfo-container';
import Navbar from './Navbar/Navbar-container';
import Today from './Today/Today-container';
import CompanyInfo from './CompanyInfo/CompanyInfo-container';
import InitialSetup from './InitialSetup';
import FooterBar from './FooterBar-container';
import SingleEvent from '../EventList/EventCard/SingleEvent-container';
import { OpeningHoursContainer as OpeningHours } from '../StoreInfo/OpeningHours-container';


import EventLoader from './Loaders/EventLoader-container';
import CategoryLoader from './Loaders/CategoryLoader-container';
import ParticipationsLoader from './Loaders/ParticipationsLoader-container';
import UploadedCategoryLogosLoader from './Loaders/UploadedCategoryLogosLoader-container';
import OpeningHoursExceptionLoader from './Loaders/OpeningHoursExceptionLoader-container';

// ******************
// Set the theme here
// ******************
import 'bulmaswatch/darkly/bulmaswatch.min.css';
import { FirebaseProfile } from '~/models/ReduxState';
import { Settings } from '~/models/Settings';

interface Props {
  dispatch: Dispatch;
  profile: FirebaseProfile;
  settings: Settings;
  location: Location;
  history: History;
}

interface State {
  redirected: boolean;
  userInfoOk: boolean;
}

export default class MainView extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.changeLanguage = this.changeLanguage.bind(this);
    this.state = { redirected: false, userInfoOk: false };
  }

  componentWillReceiveProps(nextProps: Props) {
    const wasProfileLoaded: boolean = isLoaded(this.props.profile);
    const isProfileLoaded: boolean = isLoaded(nextProps.profile);
    const isLoggedIn: boolean = isProfileLoaded && !isEmpty(nextProps.profile);
    const acceptedPrivacyPolicy = !!_.get(nextProps.profile, 'acceptedPrivacyPolicy');
    const providerEmail = _.get(nextProps.profile, 'providerData[0].email', null);
    const emailOk = (!nextProps.profile.useOtherEmail && (providerEmail || nextProps.profile.email)) || (nextProps.profile.useOtherEmail && nextProps.profile.otherEmail);
    const namesOk = !!_.get(nextProps, 'profile.firstName', false) && !!_.get(nextProps, 'profile.lastName', false);

    this.setState({ userInfoOk: !isLoggedIn || (emailOk && namesOk && acceptedPrivacyPolicy) });

    if (!this.state.redirected) {
      if (isLoaded(nextProps.profile) && isEmpty(nextProps.profile) && ['/userinfo'].indexOf(this.props.location.pathname) !== -1 && !isLoggedIn) {
        this.props.history.push('/today');
      } else if (
        isLoggedIn && (
          !nextProps.profile.firstName ||
          !nextProps.profile.lastName ||
          !emailOk ||
          !acceptedPrivacyPolicy)
      ) {
        this.props.history.push('/userinfo');
        this.setState({ redirected: true });
      } else if (wasProfileLoaded !== isProfileLoaded) {
        const currentLocation = this.props.location.pathname.substring(1);
        const landingPage = _.get(nextProps.profile, 'landingPage', 'today');
        if (_.isEmpty(currentLocation) && !_.isEmpty(landingPage)) {
          this.props.history.push(`/${landingPage}`);
        }
        this.setState({ redirected: true });
      }
    }
  }

  changeLanguage(newLanguage: string) {
    const { dispatch } = this.props;
    dispatch(setActiveLanguage(newLanguage));
    Moment.globalLocale = newLanguage;
  }

  render() {
    const { profile, settings } = this.props;
    const { userInfoOk } = this.state;

    if (isLoaded(settings) && isEmpty(settings)) {
      return <InitialSetup profile={profile} />;
    }

    return (
      <div>
        <ThemeHandler />

        <EventLoader />
        <CategoryLoader />
        <ParticipationsLoader />
        <UploadedCategoryLogosLoader />
        <OpeningHoursExceptionLoader />

        <TitleBar returnToFrontpage={() => this.props.history.push('/today')} />
        <Navbar changeLanguage={this.changeLanguage} />
        <div className="is-hidden-tablet is-centered is-fullwidth level box">
          <div className="level-item openinghours-mobile"><OpeningHours /></div>
        </div>

        {userInfoOk &&
          <Switch>
            <Route path="/today" component={Today} />
            <Route path="/event/:id" component={SingleEvent} />
            <Route path="/events" component={EventCalendar} />
            <Route path="/storeinfo" component={StoreInfo} />
            <Route path="/userinfo" component={UserInfo} />
            <Route path="/admin/tools" component={AdminTools} />
            <Route path="/admin/events" component={AdminToolsEvents} />
            <Route path="/admin/sitesettings" component={AdminSiteSettings} />
            <Route path="/login" component={Login} />
            <Route path="/register" component={Register} />
            <Route path="/companyinfo" component={CompanyInfo} />
          </Switch>
        }

        {!userInfoOk &&
          <UserInfo />
        }

        {isLoaded(settings) &&
          <FooterBar />
        }
      </div>
    );

  }
}
