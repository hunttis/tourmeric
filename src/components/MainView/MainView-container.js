import { compose } from 'redux';
import { connect } from 'react-redux';
import { firebaseConnect } from 'react-redux-firebase';

import MainView from './MainView';

export default compose(
  firebaseConnect([
    { path: '/settings' },
    { path: '/highlights' },
  ]),
  connect(state => ({
    languages: state.locale.languages,
    settings: state.firebase.data.settings,
    highlights: state.firebase.data.highlights,
  })),
  connect(({ firebase: { auth, profile } }) => ({ auth, profile })),
)(MainView);
