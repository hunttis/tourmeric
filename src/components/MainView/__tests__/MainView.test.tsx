import rrf from 'react-redux-firebase';
import React from 'react';
import { shallow } from 'enzyme';
import { createMemoryHistory, createLocation } from 'history';
import MainView from '../MainView';

import {
  mockSettings,
  mockProfile,
} from '~/components/EventList/__mocks__/mockData';
import { Settings } from '~/models/Settings';

jest.mock('~/components/AdminTools/AdminTools-container', () => 'AdminTools');
jest.mock('~/components/AdminTools/AdminToolsEvents-container', () => 'AdminToolsEvents');
jest.mock('~/components/AdminTools/SiteSettings/AdminSiteSettings-container', () => 'AdminSiteSettings');
jest.mock('~/components/AdminTools/ArticleEditor/ArticleList-container', () => 'ArticleList');

jest.mock('~/components/MainView/InitialSetup', () => 'InitialSetup');
jest.mock('~/components/MainView/FooterBar-container', () => 'FooterBar');
jest.mock('~/components/MainView/TitleBar-container', () => 'TitleBar');
jest.mock('~/components/MainView/ThemeHandler-container', () => 'ThemeHandler');

jest.mock('~/components/MainView/Account/Register-container', () => 'Register');
jest.mock('~/components/MainView/Account/Login-container', () => 'Login');
jest.mock('~/components/MainView/Navbar/Navbar-container', () => 'Navbar');
jest.mock('~/components/MainView/Today/Today-container', () => 'Today');
jest.mock('~/components/MainView/Articles/Articles-container', () => 'Articles');
jest.mock('~/components/MainView/CompanyInfo/CompanyInfo-container', () => 'CompanyInfo');

jest.mock('~/components/MainView/Loaders/ArticleLoader-container', () => 'ArticleLoader');
jest.mock('~/components/MainView/Loaders/EventLoader-container', () => 'EventLoader');
jest.mock('~/components/MainView/Loaders/CategoryLoader-container', () => 'CategoryLoader');
jest.mock('~/components/MainView/Loaders/ParticipationsLoader-container', () => 'ParticipationsLoader');
jest.mock('~/components/MainView/Loaders/UploadedCategoryLogosLoader-container', () => 'UploadedCategoryLogosLoader');
jest.mock('~/components/MainView/Loaders/OpeningHoursExceptionLoader-container', () => 'OpeningHoursExceptionLoader');
jest.mock('~/components/MainView/Loaders/UsersLoader-container', () => 'UsersLoader');
jest.mock('~/components/MainView/Loaders/HighLightsLoader-container', () => 'HighLightsLoader');
jest.mock('~/components/MainView/MainView-routes', () => 'MainViewRoutes');

jest.mock('~/components/EventList/EventCalendar/EventCalendar-container', () => 'EventCalendar');
jest.mock('~/components/EventList/EventCard/SingleEvent-container', () => 'SingleEventContainer');
jest.mock('~/components/EventList/EventCard/EventPlayerList-container', () => 'EventPlayerList');

jest.mock('~/components/StoreInfo/StoreInfo-container', () => 'StoreInfo');
jest.mock('~/components/StoreInfo/OpeningHours-container', () => 'OpeningHours');
jest.mock('~/components/UserInfo/UserInfo-container', () => 'UserInfo');


describe('MainView tests', () => {
  beforeEach(() => {
    rrf.isLoaded = jest.fn(() => true);
    rrf.isEmpty = jest.fn(() => false);
  });

  it('Renders MainView, initial setup only', () => {
    rrf.isLoaded = jest.fn(() => true);
    rrf.isEmpty = jest.fn(() => true);

    const result = shallow(
      <MainView
        settings={{} as Settings}
        dispatch={jest.fn}
        history={createMemoryHistory()}
        profile={mockProfile}
        location={createLocation('/')}
        isAdmin={false}
        initialize={jest.fn()}
        addTranslationForLanguage={jest.fn()}
      />,
    );

    expect(result).toMatchSnapshot();
  });

  it('Renders MainView, settings are not loaded', () => {
    rrf.isLoaded = jest.fn(() => false);
    rrf.isEmpty = jest.fn(() => true);

    const result = shallow(
      <MainView
        settings={mockSettings}
        dispatch={jest.fn}
        history={createMemoryHistory()}
        profile={mockProfile}
        location={createLocation('/')}
        isAdmin={false}
        initialize={jest.fn()}
        addTranslationForLanguage={jest.fn()}
      />,
    );

    expect(result).toMatchSnapshot();
  });

  it('Renders MainView, settings are loaded, ordinary logged out user', () => {

    const result = shallow(
      <MainView
        settings={mockSettings}
        dispatch={jest.fn}
        history={createMemoryHistory()}
        profile={mockProfile}
        location={createLocation('/')}
        isAdmin={false}
        initialize={jest.fn()}
        addTranslationForLanguage={jest.fn()}
      />,
    );

    expect(result).toMatchSnapshot();
  });


  it('Renders MainView, settings are loaded, admin', () => {

    const result = shallow(
      <MainView
        settings={mockSettings}
        dispatch={jest.fn}
        history={createMemoryHistory()}
        profile={mockProfile}
        location={createLocation('/')}
        isAdmin
        initialize={jest.fn()}
        addTranslationForLanguage={jest.fn}
      />,
    );

    result.setState({ userInfoOk: true });

    expect(result).toMatchSnapshot();
  });

  it('Renders MainView, settings are loaded, printpage', () => {

    const result = shallow(
      <MainView
        settings={mockSettings}
        dispatch={jest.fn}
        history={createMemoryHistory()}
        profile={mockProfile}
        location={createLocation('/print')}
        isAdmin
        initialize={jest.fn()}
        addTranslationForLanguage={jest.fn}
      />,
    );

    result.setState({ userInfoOk: true });

    expect(result).toMatchSnapshot();
  });
});
