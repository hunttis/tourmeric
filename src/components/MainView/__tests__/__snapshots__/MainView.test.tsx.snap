// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView tests Renders MainView, initial setup only 1`] = `
<InitialSetup
  profile={
    Object {
      "avatarUrl": "https://foo.com/picture",
      "dciNumber": "123512351112",
      "displayName": "Test Person",
      "email": "fake@fake.com",
      "firstName": "Test",
      "isEmpty": false,
      "isLoaded": true,
      "lastName": "Person",
      "providerData": Array [
        Object {
          "displayName": "Test Person",
          "email": "fake@fake.com",
          "photoURL": "https://foo.com/picture",
          "providerId": "facebook.com",
          "uid": "1234567890",
        },
      ],
      "role": "admin",
    }
  }
/>
`;

exports[`MainView tests Renders MainView, settings are loaded, admin 1`] = `
<div>
  <ThemeHandler />
  <EventLoader />
  <CategoryLoader />
  <ParticipationsLoader />
  <UploadedCategoryLogosLoader />
  <OpeningHoursExceptionLoader />
  <ArticleLoader />
  <HighLightsLoader />
  <UsersLoader />
  <TitleBar
    returnToFrontpage={[Function]}
  />
  <Navbar
    changeLanguage={[Function]}
  />
  <div
    className="is-hidden-tablet is-centered is-fullwidth level box"
  >
    <div
      className="level-item openinghours-mobile"
    >
      <Component />
    </div>
  </div>
  <MainViewRoutes
    isAdmin={true}
  />
  <FooterBar />
</div>
`;

exports[`MainView tests Renders MainView, settings are loaded, ordinary logged out user 1`] = `
<div>
  <ThemeHandler />
  <EventLoader />
  <CategoryLoader />
  <ParticipationsLoader />
  <UploadedCategoryLogosLoader />
  <OpeningHoursExceptionLoader />
  <ArticleLoader />
  <HighLightsLoader />
  <TitleBar
    returnToFrontpage={[Function]}
  />
  <Navbar
    changeLanguage={[Function]}
  />
  <div
    className="is-hidden-tablet is-centered is-fullwidth level box"
  >
    <div
      className="level-item openinghours-mobile"
    >
      <Component />
    </div>
  </div>
  <UserInfo />
  <FooterBar />
</div>
`;

exports[`MainView tests Renders MainView, settings are loaded, printpage 1`] = `
<div>
  <ThemeHandler />
  <EventLoader />
  <CategoryLoader />
  <ParticipationsLoader />
  <UploadedCategoryLogosLoader />
  <OpeningHoursExceptionLoader />
  <ArticleLoader />
  <HighLightsLoader />
  <UsersLoader />
  <MainViewRoutes
    isAdmin={true}
  />
</div>
`;

exports[`MainView tests Renders MainView, settings are not loaded 1`] = `
<div>
  <ThemeHandler />
  <EventLoader />
  <CategoryLoader />
  <ParticipationsLoader />
  <UploadedCategoryLogosLoader />
  <OpeningHoursExceptionLoader />
  <ArticleLoader />
  <HighLightsLoader />
  <TitleBar
    returnToFrontpage={[Function]}
  />
  <Navbar
    changeLanguage={[Function]}
  />
  <div
    className="is-hidden-tablet is-centered is-fullwidth level box"
  >
    <div
      className="level-item openinghours-mobile"
    >
      <Component />
    </div>
  </div>
  <UserInfo />
</div>
`;
